src/backends/tensorrt/plan_backend.cc

```diff
@@ -364,10 +453,62 @@ PlanBackend::CreateExecutionContext(
     }
   }
 
-  // Create CUDA stream associated with the execution context
-  const int cuda_stream_priority =
-      GetCudaStreamPriority(Config().optimization().priority());
-  RETURN_IF_ERROR(context->CreateCudaStream(cuda_stream_priority));
+  // Passing the queue for available contexts here so that completion thread
+  // knows where to inform that the context is ready for inputs.
+  std::shared_ptr<SyncQueue<size_t>> lqueue = context_queue;
+  context->completion_thread_ = std::thread([context, context_idx, lqueue]() {
+    while (true) {
+      auto OnCompletePair = context->completion_queue_.Get();
+      auto& OnComplete = OnCompletePair.first;
+      if (OnComplete == nullptr) {
+        break;
+      }
+#ifdef TRTIS_ENABLE_STATS
+      // Only need to access payload when stats is being recorded
+      auto& payloads = OnCompletePair.second;
+
+      // Only need to wait for input copy for recording stats
+      cudaEventSynchronize(context->input_ready_);
+      for (auto& payload : *payloads) {
+        if (payload.stats_ != nullptr) {
+          payload.stats_->CaptureTimestamp(
+              ModelInferStats::TimestampKind::kComputeInputEnd);
+        }
+      }
+#endif  // TRTIS_ENABLE_STATS
+
+      // The model execution associated with the OnCompletePair
+      // has consumed the inputs
```

Comment:
"... inputs. Put the context back into the available queue so that it can begin enqueuing new memcpys into the input buffers."